packages/x-components/src/composables/create-use-device.composable.ts

```diff
@@ -0,0 +1,206 @@
+          !sortedByWidthDevices.some(
+            ([, otherDeviceWidth]) =>
+              otherDeviceWidth <= windowSize.value && otherDeviceWidth > deviceWidth
+          )
+      );
+      accumulator[`is${capitalize(device)}`] = isDevice;
+      accumulator[`is${capitalize(device)}OrLess`] = computed(
+        () => deviceWidth >= windowSize.value || isDevice.value
+      );
+      accumulator[`is${capitalize(device)}OrGreater`] = computed(
+        () => deviceWidth <= windowSize.value
+      );
+      return accumulator;
+    },
+    {} as Record<string, Ref<boolean>>
+  );
+}
+
+/**
+ * A function that returns the current device orientation as a reactive value.
+ *
+ * @returns A reactive value indicating the current device
+ * orientation.
+ *
+ * @internal
+ */
+function getOrientation(): UseDeviceReturn['orientation'] {
+  const { orientation } = useScreenOrientation();
+  return computed(() => {
+    if (['landscape-primary', 'landscape-secondary'].includes(orientation.value ?? '')) {
+      return 'landscape';
+    } else if (['portrait-primary', 'portrait-secondary'].includes(orientation.value ?? '')) {
+      return 'portrait';
+    }
+    return undefined;
+  });
+}
+
+/**
+ * A function that returns a reactive boolean indicating whether the current device is
+ * touch-enabled.
+ *
+ * @returns A reactive boolean indicating whether the current device is touch-enabled.
+ *
+ * @internal
+ */
+function getIsTouchable(): Ref<boolean> {
+  const isTouchableRef = ref(detectTouchable());
+  if (window) {
+    useEventListener(window, 'resize', () => (isTouchableRef.value = detectTouchable()), {
+      passive: true
+    });
+  }
+  return isTouchableRef;
+}
+
+/**
+ * A function that returns a reactive string indicating the name of the currently detected device
+ * based on the provided devices and device flags.
+ *
+ * @param  devices - An object containing the breakpoints, where the key is the name of the device
+ * and the value is the screen width.
+ * @param devicesFlags - An object containing multiple reactive flags and values for detecting
+ * the current device.
+ * @returns A reactive string indicating the name of the currently detected device.
+ *
+ * @internal
+ */
+function getDeviceName(
+  devices: Record<string, number>,
+  devicesFlags: UseDeviceFlags<string>
+): Ref<string> {
+  return computed(
+    () =>
+      Object.keys(devices).find(device => devicesFlags[`is${capitalize(device)}` as any]?.value) ??
+      ''
+  );
+}
+
+/**
+ * A utility function that detects whether the current device is touch-enabled.
+ *
+ * @returns A boolean indicating whether the current device is touch-enabled.
+ *
+ * @internal
+ */
+function detectTouchable(): boolean {
+  return 'ontouchstart' in window || navigator.maxTouchPoints > 0;
+}
+
+/**
+ * Util function to capitalize a string.
+ *
+ * @param str - The string to capitalize.
+ * @returns The string capitalized.
+ *
+ * @internal
+ */
+function capitalize(str: string): string {
```

Comment:
We already have a util for this inside x-components.

Reply:
I was looking for it in the `x-utils` package. We have to move it there in the near future.